Source/Core/Common/ENet.cpp

```diff
@@ -18,7 +18,7 @@ void WakeupThread(ENetHost* host)
     address.port = host->address.port;
   else
     enet_socket_get_address(host->socket, &address);
-  address.host = 0x0100007f;  // localhost
+  address.host = in6addr_loopback;  // localhost or ::1
```

Comment:
I see the 6 in there, but I'm not sure if this is an IPv4 or IPv6 address in the end (vs. it simply being an IPv6-capable oblivious thing, no matter what is used in the end). Is this smart enough to do the right thing on a system that doesn't have IPv6 (and also, on a system that only has IPv6, if those exist)?

Reply:
in6addr_loopback is ::1, the IPv6 localhost address. It will fail on machines where the kernel doesn't support IPv6 at all but it should work just fine if the kernel supports IPv6 and it's just disabled in the network or not supported by the ISP.